core/includes/module.inc

```diff
@@ -379,15 +379,21 @@ function module_load_all_includes($type, $name = NULL) {
  * - Invoke hook_modules_installed().
  * - Invoke hook_modules_enabled().
  *
- * @param string[] $module_list
+ * @param string $module_list
  *   An array of module names.
  * @param bool $enable_dependencies
- *   If TRUE, dependencies will automatically be added and enabled in the
- *   correct order. This incurs a significant performance cost, so use FALSE
+ *   If TRUE, dependency-checking is enabled. This means that if dependencies
+ *   are available, they will be automatically added and enabled in the correct
+ *   order. If they are unavailable, the module(s) will not be enabled.
+ *   Note that dependency-checking incurs a significant performance cost. If
+ *   FALSE, dependency-checking is bypassed and the module(s) will be enabled
+ *   regardless of whether or not their dependencies are available or listed in
+ *   the correct order. This could lead to problems, and so should only be used
```

Comment:
I like this change 👍, but can we please split the "If TRUE..." and "If FALSE..." into two separate paragraphs, or bullet points? (for better readability)

Reply:
I'm not sure you can have separate paragraphs within `@param` blocks, since a blank line indicates the end of the `@param` section... So I updated it to use bullets instead.